packages/control-property-editor/src/App.tsx

```diff
@@ -38,8 +42,13 @@ export default function App(appProps: AppProps): ReactElement {
             sheet.cssRules.length
         );
     }, []);
-    const { t } = useTranslation();
-    const dispatch = useAppDispatch();
+
+    useEffect(() => {
+        if (scenario) {
+            dispatch(setProjectScenario(scenario));
```

Comment:
Wouldn't it be simpler to set the scenario in the `start` function? As far as I know it should only happen at the start of the application and we have store with dispatch available there.

Reply:
We cannot set it there because it is not wrapped in a Provider and the store is not initialized. The first component that can do that is the App.tsx.
<img width="679" alt="Screenshot 2024-01-11 at 15 40 10" src="https://github.com/SAP/open-ux-tools/assets/39880364/07aeadba-8291-457d-808c-098c2987a978">

Reply:
Moreover, the `start` function is not a function component but a method you call to start rendering the application. And by the rules of react you can only use hooks when:
- ✅ Calling them at the top level in the body of a [function component](https://react.dev/learn/your-first-component).
- ✅ Calling them at the top level in the body of a [custom Hook](https://react.dev/learn/reusing-logic-with-custom-hooks).

Reply:
The point was to avoid hooks all together. This is what I had in mind, but you are right the store is not yet ready there and we would probably not get much simpler solution.
![image](https://github.com/SAP/open-ux-tools/assets/6721493/7425f12a-3ba1-4230-b94c-7fcff18eafb8)

Reply:
Never mind, somehow I had not saved the changes which was the issue. The change above actually works fine. Can we do that instead of using hooks in the react component? That seems more straight forward solution for setting something once at the start up.
<img width="392" alt="image" src="https://github.com/SAP/open-ux-tools/assets/6721493/0369e505-d74d-484c-8bf2-b8ae918ec61e">

Reply:
Changed in: https://github.com/SAP/open-ux-tools/pull/1553/commits/39c818755642cc6f06ad90b77692868cc6ee01f7